### PR TITLE
Implement skip_serializing for enum variant

### LINKED ISSUE #638: Attribute skip_(de)serializing for enum variants
```rust
#[derive(Serialize, Deserialize)]
enum E {
    A,
    B,
    #[serde(skip_serializing, skip_deserializing)]
    C,
}
```

When serializing, it should error when the value is that variant. When deserializing, it should be like the variant doesn't exist.

### PR DESCRIPTION
**Pull request by shinglyu**
Fixes #638. Happy Holidays!

### CODE REVIEW

File: testing/tests/test_ser.rs [hunk 1]
```diff
@@ -309,6 +317,8 @@ declare_ser_tests! {
                 Token::I32(2),
             Token::EnumMapEnd,
         ],
+        //Enum::Skipped=> &[Token::EnumUnit("Enum", "Skipped")], //should fail
```
Comment (oli-obk) [inline testing/tests/test_ser.rs]: Why are these commented out?

Comment (shinglyu) [inline testing/tests/test_ser.rs]: Sorry, these should be removed.

**Comment by dtolnay** [comment]
Very nice once again! I hope you are interested in continuing to contribute to Serde, it's been a big help.

**Comment by shinglyu** [comment]
@dtolnay My work on [servo](https://servo.org) and RustPython (https://github.com/shinglyu/RustPython) depends heavily on serde. I'll try to fix whatever I stumbled upon. :)

**Pull request merged**